torch/utils/data/datapipes/utils/common.py

```diff
@@ -73,30 +74,44 @@ def deprecation_warning_torchdata(name):
                   f"Please import those features from the new package TorchData: https://github.com/pytorch/data",
                   DeprecationWarning)
 
+
 class StreamWrapper:
-    def __init__(self, file_obj):
-        self.file_obj = file_obj
+    pass
+
+
+def wrap_stream(file_obj):
+    return get_wrapper_cls(type(file_obj))(file_obj)
+
+
+def _magic_fn_wrapper(fn_name):
+    def magic_fn(self, *args, **kwargs):
+        return getattr(self.file_obj, fn_name)(*args, **kwargs)
+    return magic_fn
 
-    def __getattr__(self, name):
-        return getattr(self.file_obj, name)
 
-    def __del__(self):
-        self.file_obj.close()
+@cache
```

Comment:
This is introduced from python 3.9.
We can either use `lru_cache` or implement own cache inside the `get_wrapper_cls` with a dict.